pkg/controller/release/strategy_executor.go

```diff
@@ -0,0 +1,470 @@
+
+		if targetStep >= int32(len(strategy.Steps)) {
+			err := fmt.Errorf("no step %d in strategy for Release %q",
+				targetStep, controller.MetaKey(curr.release))
+			return PipelineBreak, nil, nil, shippererrors.NewUnrecoverableError(err)
+		}
+
+		if ready, clusters := checkInstallation(curr); !ready {
+			if len(curr.installationTarget.Spec.Clusters) != len(curr.installationTarget.Status.Clusters) {
+				cond.SetUnknown(
+					shipper.StrategyConditionContenderAchievedInstallation,
+					conditions.StrategyConditionsUpdate{
+						Step:               targetStep,
+						LastTransitionTime: time.Now(),
+					},
+				)
+			} else {
+				cond.SetFalse(
+					shipper.StrategyConditionContenderAchievedInstallation,
+					conditions.StrategyConditionsUpdate{
+						Reason:             ClustersNotReady,
+						Message:            fmt.Sprintf("clusters pending installation: %v. for more details try `kubectl describe it %s`", clusters, curr.installationTarget.Name),
+						Step:               targetStep,
+						LastTransitionTime: time.Now(),
+					},
+				)
+			}
+
+			return PipelineBreak, e.buildContenderStrategyConditionsPatch(cond, targetStep, isLastStep, e.hasIncumbent), nil, nil
+		}
+
+		cond.SetTrue(
+			shipper.StrategyConditionContenderAchievedInstallation,
+			conditions.StrategyConditionsUpdate{
+				LastTransitionTime: time.Now(),
+				Step:               targetStep,
+			},
+		)
+
+		return PipelineContinue, nil, nil, nil
+	}
+}
+
+func genCapacityEnforcer(curr, succ *releaseInfo) PipelineStep {
+	return func(e *StrategyExecutor, cond conditions.StrategyConditionsMap) (PipelineContinuation, []ExecutorResult, []ReleaseStrategyStateTransition, error) {
+		var targetStep, capacityWeight int32
+		var strategy *shipper.RolloutStrategy
+		var strategyStep shipper.RolloutStrategyStep
+		var condType shipper.StrategyConditionType
+
+		isHead := succ == nil
+
+		if isHead {
+			targetStep = curr.release.Spec.TargetStep
+			strategy = curr.release.Spec.Environment.Strategy
+			strategyStep = strategy.Steps[targetStep]
+			capacityWeight = strategyStep.Capacity.Contender
+			condType = shipper.StrategyConditionContenderAchievedCapacity
+		} else {
+			targetStep = succ.release.Spec.TargetStep
+			strategy = succ.release.Spec.Environment.Strategy
+			strategyStep = strategy.Steps[targetStep]
+			capacityWeight = strategyStep.Capacity.Incumbent
+			condType = shipper.StrategyConditionIncumbentAchievedCapacity
+		}
+
+		if targetStep >= int32(len(strategy.Steps)) {
+			err := fmt.Errorf("no step %d in strategy for Release %q",
+				targetStep, controller.MetaKey(curr.release))
+			return PipelineBreak, nil, nil, shippererrors.NewUnrecoverableError(err)
+		}
+
+		isLastStep := int(targetStep) == len(strategy.Steps)-1
+
+		if achieved, newSpec, clustersNotReady := checkCapacity(curr.capacityTarget, uint(capacityWeight)); !achieved {
+			e.info("release %q hasn't achieved capacity yet", curr.release.Name)
```

Comment:
`info` already prints out the name of the current release by default, so there's no need to say `release %q has blah...` here and a few other places anymore, as that was only necessary when we were operating on two releases at once.